js/main.js

```diff
@@ -169,3 +186,50 @@ function handleEditClick(event) {
   $image.setAttribute('src', data.editing.photoAddress);
 }
 $unorderedList.addEventListener('click', handleEditClick);
+
+// Confirmation modal when the user clicks delete entry
+var $deleteLink = document.querySelector('.delete-link');
+function handleDeleteClick(event) {
+  document.querySelector('.bg-modal').style.display = 'flex';
+}
+$deleteLink.addEventListener('click', handleDeleteClick);
+
+// Hide the modal if the user clicks cancel
+var $cancelClick = document.querySelector('.cancel-button');
+function handleCancelClick(event) {
+  document.querySelector('.bg-modal').style.display = 'none';
+}
+$cancelClick.addEventListener('click', handleCancelClick);
+
+// Remove the entry from the data model and the entry's DOM tree
+// from the page if the user clicks delete.
+var $confirmClick = document.querySelector('.confirm-button');
+function handleConfirmClick(event) {
+// remove the modal window
+  document.querySelector('.bg-modal').style.display = 'none';
+  // splice the entry out of the data.entries array
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.entries[i].entryId === data.editing.entryId) {
+      data.entries.splice(i, 1);
+
+    }
+  }
+  var $listItems = document.querySelectorAll('li');
+  for (var j = 0; j < $listItems.length; j++) {
+    var $listItemsId = $listItems[j].getAttribute('data-entry-id');
+    var $idInteger = parseInt($listItemsId);
+    if ($idInteger === data.editing.entryId) {
+      $listItems[j].remove();
+    }
+  }
```

Comment:
If there are no entries left, be sure to show the "no entries" message too.

Reply:
https://user-images.githubusercontent.com/96929704/154734912-4b5be06c-2893-4b03-bbbb-ddc2881c3d69.mov